account_anglo_saxon_stock_move/model/invoice.py

```diff
@@ -43,87 +43,70 @@ def line_get_convert(self, line, part, date):
 
     @api.multi
     def action_cancel(self):
-        aml_obj = self.env['account.move.line']
-        aml_ids = []
-        for inv in self:
-            if not inv.move_id:
-                continue
-            for aml_brw in inv.move_id.line_id:
-                # Unreconcile all non-receivable and non-payable lines
-                if aml_brw.account_id.reconcile and aml_brw.account_id.type \
-                        not in ('receivable', 'payable'):
-                    aml_ids.append(aml_brw.id)
-        if aml_ids:
-            aml_obj._remove_move_reconcile(aml_ids)
-
+        for inv in self.filtered(lambda inv: inv.move_id):
+            amr_ids = inv.invoice_line.mapped('move_id.aml_ids.reconcile_id')
```

Comment:
I saw too much changes from original to this one.
Are you sure that is working with the same behaviour?

If is yes: 👍 Excellent!